ddtrace/internal/flare.py

```diff
@@ -0,0 +1,184 @@
+                    self._clean_up_files()
+
+    def _generate_config_file(self, pid: int):
+        config_file = f"{TRACER_FLARE_DIRECTORY}/tracer_config_{pid}.json"
+        try:
+            with open(config_file, "w") as f:
+                tracer_configs = {
+                    "configs": config.__dict__,
+                }
+                json.dump(
+                    tracer_configs,
+                    f,
+                    default=lambda obj: obj.__repr__() if hasattr(obj, "__repr__") else obj.__dict__,
+                    indent=4,
+                )
+        except Exception as e:
+            log.warning("Failed to generate %s: %s", config_file, e)
+            if os.path.exists(config_file):
+                os.remove(config_file)
+
+    def _revert_configs(self):
+        ddlogger = get_logger("ddtrace")
+        handler_to_remove = _get_handler(ddlogger, TRACER_FLARE_FILE_HANDLER_NAME)
+        if handler_to_remove:
+            ddlogger.removeHandler(handler_to_remove)
+            log.debug("ddtrace logs will not be routed to the %s file handler anymore", TRACER_FLARE_FILE_HANDLER_NAME)
+        else:
+            log.debug("Could not find %s to remove", TRACER_FLARE_FILE_HANDLER_NAME)
+        ddlogger.setLevel(self.original_log_level)
+
+    def _generate_payload(self, params: Dict[str, str]) -> Tuple[dict, bytes]:
+        tar_stream = io.BytesIO()
+        with tarfile.open(fileobj=tar_stream, mode="w") as tar:
+            for file_name in os.listdir(TRACER_FLARE_DIRECTORY):
+                flare_file_name = f"{TRACER_FLARE_DIRECTORY}/{file_name}"
+                tar.add(flare_file_name)
+        tar_stream.seek(0)
+
+        newline = "\r\n"
+        boundary = secrets.token_hex(16)
+        body = io.StringIO()
+        for key, value in params.items():
+            body.write("--" + boundary + newline)
+            body.write(f'Content-Disposition: form-data; name="{key}"{newline}{newline}')
+            body.write(f"{value}{newline}")
+
+        body.write("--" + boundary + newline)
+        body.write(f'Content-Disposition: form-data; name="flare_file"; filename="flare.tar"{newline}')
+        body.write(f"Content-Type: application/octet-stream{newline}{newline}")
+        body.write(tar_stream.getvalue().decode() + newline)
+        body.write("--" + boundary + "--")
+        headers = {
+            "Content-Type": "multipart/form-data; boundary=" + boundary,
+            "Content-Length": str(len(body.getvalue().encode())),
+        }
+        if config._dd_api_key:
+            headers["DD-API-KEY"] = config._dd_api_key
+        return headers, body.getvalue().encode()
+
+    def _get_valid_logger_level(self, flare_log_level: int) -> int:
+        valid_original_level = 100 if self.original_log_level == 0 else self.original_log_level
+        return min(valid_original_level, flare_log_level)
+
+    def _clean_up_files(self):
+        try:
+            shutil.rmtree(TRACER_FLARE_DIRECTORY)
```

Comment:
should you add a `return` after this?

if we delete the directory then we'll also delete the lock file, so the next `os.remove(TRACER_FLARE_LOCK)` should always fail?